libV2/schemaUtils.js

```diff
@@ -1054,16 +1054,150 @@ let QUERYPARAM = 'query',
     return HEADER_TYPE.INVALID;
   },
 
-  resolveRequestBodyData = (context, requestBodySchema, bodyType) => {
+  /**
+   * Gets XML Example data in correct format based on schema
+   *
+   * @param {Object} context - Global context object
+   * @param {Object} exampleData - Example data to be used
+   * @param {Object} requestBodySchema - Schema of the request body
+   * @returns {String} XML Example data
+   */
+  getXMLExampleData = (context, exampleData, requestBodySchema) => {
+    const { parametersResolution, indentCharacter } = context.computedOptions;
+
+    let reqBodySchemaWithExample = requestBodySchema;
+
+    // Assign example at schema level to be faked by xmlSchemaFaker
+    if (typeof requestBodySchema === 'object') {
+      reqBodySchemaWithExample = Object.assign({}, requestBodySchema, { example: exampleData });
+    }
+
+    return xmlFaker(null, reqBodySchemaWithExample, indentCharacter, parametersResolution);
+  },
+
+  /**
+   * Generates postman equivalent examples which contains request and response mappings of
+   * each example based on examples mentioned ind definition
+   *
+   * @param {Object} context - Global context object
+   * @param {Object} responseExamples - Examples defined in the response
+   * @param {Object} requestBodyExamples - Examples defined in the request body
+   * @param {Object} responseBodySchema - Schema of the response body
+   * @param {Boolean} isXMLExample - Whether the example is XML example
+   * @returns {Array} Examples for corresponding operation
+   */
+  generateExamples = (context, responseExamples, requestBodyExamples, responseBodySchema, isXMLExample) => {
+    const pmExamples = [];
+
+    _.forEach(responseExamples, (responseExample, index) => {
+
+      if (!_.isObject(responseExample)) {
+        return;
+      }
+
+      let responseExampleData = getExampleData(context, { [responseExample.key]: responseExample.value }),
+        requestExample;
+
+      if (isXMLExample) {
+        responseExampleData = getXMLExampleData(context, responseExampleData, responseBodySchema);
+      }
+
+      if (_.isEmpty(requestBodyExamples)) {
+        pmExamples.push({
+          response: responseExampleData,
+          name: _.get(responseExample, 'value.summary') || responseExample.key
+        });
+        return;
+      }
+
+      requestExample = _.find(requestBodyExamples, (example, index) => {
+        if (example.key === responseExample.key) {
+          requestBodyExamples[index].isUsed = true;
+          return true;
+        }
+        return false;
+      });
+
+      if (!requestExample) {
+        if (requestBodyExamples[index] && !requestBodyExamples[index].isUsed) {
+          requestExample = requestBodyExamples[index];
+          requestBodyExamples[index].isUsed = true;
```

Comment:
@VShingala This line is not needed.

Reply:
@webholik So basically, we're marking an example is used here so we don't use if again.